### PR TITLE
view constness and cleanup

### PR DESCRIPTION
**Pull request by McCallisterRomer**
* Removed Registry::ViewAll() and replaced usage with views
* Fixed views not working with const template arguments and const Registry*
* Added support for view<Entity>.

### CODE REVIEW

File: nc/include/ecs/view.h [hunk 1]
```diff
@@ -1,80 +1,70 @@
...
              */
-            auto rend() noexcept
-            {
-                return std::make_reverse_iterator(begin());
-            }
+            auto rend() const noexcept { return std::make_reverse_iterator(begin()); }
 
             /**
-             * @brief Returns the number of components in the view.
+             * @brief Returns the number of element in the view.
```
Comment (jaremieromer) [inline nc/include/ecs/view.h]: elements

File: nc/source/ui/editor/EditorControls.h [hunk 2]
```diff
@@ -247,13 +248,13 @@ namespace nc::ui::editor::controls
...
-        ComponentSystemHeader<PhysicsBody>("Physics Body", registry->ViewAll<PhysicsBody>());
-        ComponentSystemHeader<Transform>("Transform", registry->ViewAll<Transform>());
-        ComponentSystemHeader<nc::MeshRenderer>("Mesh Renderer", registry->ViewAll<MeshRenderer>());
-        ComponentSystemHeader<nc::PointLight>("Point Light", registry->ViewAll<PointLight>());
+        ComponentSystemHeader<Collider>("Collider", view<Collider>(registry));
+        ComponentSystemHeader<NetworkDispatcher>("NetworkDispatcher", view<NetworkDispatcher>(registry));
+        ComponentSystemHeader<ParticleEmitter>("Particle Emitter", view<ParticleEmitter>(registry));
+        ComponentSystemHeader<PhysicsBody>("Physics Body", view<PhysicsBody>(registry));
+        ComponentSystemHeader<Transform>("Transform", view<Transform>(registry));
+        ComponentSystemHeader<nc::MeshRenderer>("Mesh Renderer", view<MeshRenderer>(registry));
```
Comment (jaremieromer) [inline nc/source/ui/editor/EditorControls.h]: do we still need these nc:: namespaces?

**Review by jaremieromer — APPROVED** [review]

File: nc/source/ui/editor/EditorControls.h (hunk 2 above)
Comment (McCallisterRomer) [reply nc/source/ui/editor/EditorControls.h]: Nope, don't event know why they're there.

File: nc/include/ecs/view.h (hunk 1 above)
Comment (McCallisterRomer) [reply nc/include/ecs/view.h]: earth, wind...

**Pull request merged**